src/picongpu/include/particles/creation/CreatorBase.hpp

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright 2015 Heiko Burau, Marco Garten
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "simulation_defines.hpp"
+#include "traits/Resolve.hpp"
+#include "mappings/kernel/AreaMapping.hpp"
+
+#include "fields/FieldB.hpp"
+#include "fields/FieldE.hpp"
+
+#include "compileTime/conversion/TypeToPointerPair.hpp"
+#include "memory/boxes/DataBox.hpp"
+
+namespace picongpu
+{
+namespace particles
+{
+namespace creation
+{
+
+/** This functor is a base class for an actual particle creator.
+ *
+ * \tparam T_DestSpecies electron species to be created
+ */
+template<typename T_SourceSpecies, typename T_TargetSpecies>
+struct CreatorBase
+{
+
+    typedef T_SourceSpecies SourceSpecies;
+    typedef T_TargetSpecies TargetSpecies;
+
+    typedef typename SourceSpecies::FrameType FrameType;
+
```

Comment:
please add an `PMACC_CASSRT_...` and check if species has the flag `interpolation<>`